packages/checkout/widgets-lib/src/widgets/sale/hooks/useCurrency.ts

```diff
@@ -0,0 +1,58 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { useState, useEffect } from 'react';
+import { PRIMARY_SALES_API_BASE_URL } from '../utils/config';
+
+type CurrencyResponse = {
+  decimals: number;
+  erc20_address: string;
+  name: string;
+};
+
+type UseCurrencyParams = {
+  env: string;
+  environmentId: string;
+  currencyName?: string;
+};
+
+export const useCurrency = ({
+  env,
+  environmentId,
+  currencyName,
+}: UseCurrencyParams) => {
+  const [currencyResponse, setCurrencyResponse] = useState<
+  CurrencyResponse | undefined
+  >(undefined);
+
+  useEffect(() => {
+    const fetchCurrency = async () => {
```

Comment:
before this line you should consider checking if `environmentId` or `env` are undefined. if so fail early